go/enclave/nodetype/sequencer.go

```diff
@@ -186,7 +186,12 @@ func (s *sequencer) createNewHeadBatch(l1HeadBlock *common.L1Block) error {
 	}
 
 	// todo - time is set only here; take from l1 block?
-	if _, err := s.produceBatch(sequencerNo.Add(sequencerNo, big.NewInt(1)), l1HeadBlock.Hash(), headBatch.Hash(), transactions, uint64(time.Now().Unix())); err != nil {
+	if _, err := s.produceBatch(sequencerNo.Add(sequencerNo, big.NewInt(1)), l1HeadBlock.Hash(), headBatch.Hash(), transactions, uint64(time.Now().Unix()), skipBatchIfEmpty); err != nil {
+		if errors.Is(err, components.ErrNoTransactionsToProcess) {
+			// skip batch production when there are no transactions to process
+			s.logger.Info("Skipping batch production, no transactions to execute")
```

Comment:
we might want to eventually make this a DEBUG, as we'll see quite a lot of it

Reply:
Yeah 100%, it will be implied by the silence I guess. But yeah I'll leave it there for now just to make sure it's behaving as expected until we're confident.